core/temp/temp_driver.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file TempDriver.cpp
+ * @author Mateusz Krajewski (matikrajek42@gmail.com)
+ * @brief 
+ * @version 0.1
+ * @date 2024-11-07
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#include <optional>
+#include <vector>
+#include <cstdint>
+#include <filesystem>
+#include <string>
+#include <fstream>
+#include <stdexcept>
+#include "ara/log/log.h"
+#include "ara/core/result.h"
+#include "core/temp/temp_driver.hpp"
+#include "core/common/condition.h"
+#include "ara/com/com_error_domain.h"
+
+namespace simba {
+namespace core {
+namespace temp {
+
+namespace {
+    constexpr const char* kSensorPath = "/sys/bus/w1/devices/";
+    constexpr auto kManagerName = "w1_bus_master1";
+}
+TempDriver::TempDriver() {
+     ara::log::LoggingMenager::GetInstance()->CreateLogger("temp", "", ara::log::LogLevel::kInfo);
+}
+ara::core::Result<double> TempDriver::ReadTemp(const std::string& sensorPhysicalID) {
+    std::ifstream file(kSensorPath + sensorPhysicalID + "/temperature");
+    if (!file) {
+        ara::log::LogDebug() << "Sensor " << sensorPhysicalID << " not available!";
+        return ara::com::MakeErrorCode(
+            ara::com::ComErrc::kErroneousFileHandle, "Can't open file for 1Wire sensor");
+    }
+    std::string line;
+    std::getline(file, line);
+    try {
+        double value = std::stoi(line) / 1000.0;
```

Comment:
```suggestion
        const double value = static_cast<double>(std::stoi(line)) / 1000.0;
```

Potrzebujemy takiej precyzji ?

Reply:
moglby byc float bo to jest i tak 9bit